src/sfdxProject.ts

```diff
@@ -98,7 +180,13 @@ export class SfdxProject {
    * **Throws** *{@link SfdxError}{ name: 'InvalidProjectWorkspace' }* If the current folder is not located in a workspace.
    */
   public static async resolve(path?: string): Promise<SfdxProject> {
-    return new SfdxProject(await this.resolveProjectPath(path));
+    const _path = path || process.cwd();
+    if (!SfdxProject.instances.has(_path)) {
+      const project = new SfdxProject(await this.resolveProjectPath(_path));
+      SfdxProject.instances.set(_path, project);
+    }
+    // @ts-ignore Because of the pattern above this is guaranteed to return an instance
+    return SfdxProject.instances.get(_path);
```

Comment:
`ensureInstance` wouldn't work here due to the private constructor for SfdxProject.